packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -0,0 +1,275 @@
+      else {
+        self.logger.info("Could not fetch metadata for '\(imageUrl.absoluteString)'.")
+        return completion(nil)
+      }
+      let exif = self.readExifFrom(imageMetadata:properties)
+      return completion(exif)
+    }
+  }
+  
+  private func readExifFrom(imageMetadata: [String: Any]) -> [String: Any] {
+    var exif = imageMetadata[kCGImagePropertyExifDictionary as String] as! [String: Any]? ?? [:]
+
+    // Copy ["{GPS}"]["<tag>"] to ["GPS<tag>"]
+    let gps = imageMetadata[kCGImagePropertyGPSDictionary as String] as! [String: Any]?
+    if (gps != nil) {
+      gps!.forEach { (key, value) in
+        exif["GPD\(key)"] = value
+      }
+    }
+
+    // Inject orientation into exif
+    let orientationKey = kCGImagePropertyOrientation as String
+    let orientationValue = imageMetadata[orientationKey]
+    if (orientationValue != nil) {
+      exif[orientationKey] = orientationValue
+    }
+
+    return exif
+  }
+
+  // MARK: Video
+
+  // TODO: convert to async/await syntax once we drop support for iOS 12
+  func handleVideo(mediaInfo: MediaInfo, completion: (Result) -> Void) {
+    guard let pickedVideoUrl = mediaInfo[.mediaURL] as! URL?
+                            ?? mediaInfo[.referenceURL] as! URL? else { return completion(.Failure(FailedToOpenVideoError())) }
+
+    let targetPath = self.generateDestinationPath(withExtension: ".mov")
+    let targetUrl: URL = URL.init(fileURLWithPath: targetPath)
+
+    do {
+      // we copy the file as `moveItem(at:,to:)` throws an error in iOS 13 due to missing permissions
+      try FileManager.default.copyItem(at: pickedVideoUrl, to: targetUrl)
+    } catch {
+      return completion(.Failure(FailedToPickVideo(reason: error)))
+    }
+
+    // Adding information about asset
+    let asset = AVURLAsset.init(url: targetUrl)
+    guard let size: CGSize = asset.tracks(withMediaType: .video).first?.naturalSize else {
+      return completion(.Failure(FailedToReadVideoSize()))
+    }
+
+    // If video was edited (the duration is affected) then read the duration from the original edited video.
+    // Otherwise read the duration from the target video file.
+    // TODO: (@bbarthec): inspect whether it makes sense to read duration from two different assets
+    let videoAssetToReadDurationFrom = self.pickingOptions.allowsEditing ? AVURLAsset.init(url: pickedVideoUrl) : asset
+    let duration = self.readVideoAssetDuration(videoAssetToReadDurationFrom)
+
+    let result: Response = .Video(VideoInfo(uri: targetUrl.absoluteString,
+                                            width: size.width,
+                                            height: size.height,
+                                            duration: duration))
+    completion(.Success(result))
+  }
+
+  private func readVideoAssetDuration(_ asset: AVURLAsset) -> Double {
+    return Double(asset.duration.value) / Double(asset.duration.timescale) * 1000 // miliseconds
```

Comment:
https://realm.github.io/SwiftLint/number_separator.html
```suggestion
    return Double(asset.duration.value) / Double(asset.duration.timescale) * 1_000 // miliseconds
```